File: inferago_complete/frontend/src/api/review.ts
import api from "./client";

export interface Review {
  id: string;
  tenant_id: string;
  run_id: string;
  workflow_id: string;
  user_id: string;
  status: "pending" | "in_review" | "approved" | "blocked" | "escalated" | "completed";
  decision: string | null;
  max_risk_score: number;
  finding_count: number;
  top_finding_severity: string | null;
  governing_policy_id: string | null;
  governing_policy_name: string | null;
  decision_by: string | null;
  decision_at: string | null;
  decision_note: string | null;
  created_at: string;
  updated_at: string;
}

export interface ReviewQueueStats {
  total: number;
  pending: number;
  in_review: number;
  approved: number;
  blocked: number;
  escalated: number;
  avg_risk_score: number;
}

export const getReviews = (params?: {
  status?: string;
  workflow_id?: string;
  min_risk?: number;
  limit?: number;
}): Promise<Review[]> =>
  api.get("/api/reviews", { params }).then((r) => r.data);

export const getReview = (id: string): Promise<Review> =>
  api.get(`/api/reviews/${id}`).then((r) => r.data);

export const getReviewStats = (): Promise<ReviewQueueStats> =>
  api.get("/api/reviews/stats").then((r) => r.data);

export const decideReview = (
  id: string,
  payload: { status: string; decision_note?: string }
): Promise<Review> =>
  api.patch(`/api/reviews/${id}`, payload).then((r) => r.data);

File: inferago_complete/backend/app/schemas/review.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
import uuid


# ── Request ────────────────────────────────────────────────────────────────────

class ReviewUpdate(BaseModel):
    """Payload for a reviewer making a decision on a review item."""
    status: str                            # approved | blocked | escalated
    decision_note: Optional[str] = None


# ── Response ──────────────────────────────────────────────────────────────────

class ReviewResponse(BaseModel):
    id                    : uuid.UUID
    tenant_id             : uuid.UUID
    run_id                : uuid.UUID
    workflow_id           : uuid.UUID
    user_id               : uuid.UUID
    status                : str
    decision              : Optional[str]
    max_risk_score        : int
    finding_count         : int
    top_finding_severity  : Optional[str]
    governing_policy_id   : Optional[uuid.UUID]
    governing_policy_name : Optional[str]
    decision_by           : Optional[uuid.UUID]
    decision_at           : Optional[datetime]
    decision_note         : Optional[str]
    created_at            : datetime
    updated_at            : datetime
    model_config = {"from_attributes": True}


class ReviewQueueStats(BaseModel):
    total          : int
    pending        : int
    in_review     : int
    approved       : int
    blocked        : int
    escalated      : int
    avg_risk_score : float
